File: 200714.py
arr_in = [1,0,2,3,0,4,5,0]

class Solution:
    def duplicateZeros(self, arr):
        new_arr = []
        
        for i in arr:
            new_arr.append(i)
            if i == 0:
                new_arr.append(i)

        # This works!
        for j in range(len(arr)):
            arr[j] = new_arr[j]
        
        # But this doesn't!
        #arr = new_arr[0:len(arr)]


ans = Solution()
ans.duplicateZeros(arr_in)


print(arr_in)

File: 200714.js
var arr = [1,0,2,3,0,4,5,0]

// function solution(myArr) {
//     let newArr = []
//     for (let i = 0; i < myArr.length; i++) {
//         newArr.push(myArr[i])
//         if (myArr[i]==0) {
//             newArr.push(0)
//         }
//     }

//     for (let i = 0; i < myArr.length; i++) {
//         myArr[i] = newArr[i]
//     }
// }

var duplicateZeros = function(myArr) {
    let newArr = myArr.slice()
    
    let i = 0;
    let j = 0;

    while (j < myArr.length) {
        myArr[j] = newArr[i];
        console.log(myArr);
        console.log(newArr);
        if (newArr[i] == 0) {
            myArr[j+1] = 0;
            j+=2;
            i++;
        }
        j++;
        i++;    
    }

}

duplicateZeros(arr)
console.log(arr)